CONTRIBUTING.md

```diff
@@ -0,0 +1,146 @@
+# Hyperion Contributing Guide
+
+* [New Contributor Guide](#contributing-guide)
+  * [Ways to Contribute](#ways-to-contribute)
+  * [Find an Issue](#find-an-issue)
+  * [Ask for Help](#ask-for-help)
+  * [Pull Request Lifecycle](#pull-request-lifecycle)
+  * [Development Environment Setup](#development-environment-setup)
+  * [Pull Request Checklist](#pull-request-checklist)
+
+Hello there! We are glad that you want to contribute to our project! 💖
+
+As you get started, you are in the best position to give us feedback on areas of
+our project that we need help with including:
+
+* Problems found during setting up a new developer environment
+* Gaps in our Quickstart Guide or documentation
+* Bugs in our automation scripts
+
+If anything doesn't make sense, or doesn't work when you run it, please open a
+bug report and let us know!
+
+## Ways to Contribute
+
+
+We welcome many different types of contributions including:
+
+* New features
+* Builds, CI/CD
+* Bug fixes
+* Documentation
+* Issue Triage
+
+Not everything happens through a GitHub pull request. Please go to our [contact us](https://www.sigscalr.io/contact.html) and let's discuss how we can work together. 
+
+
+## Find an Issue
+
+We have good first issues for new contributors and help wanted issues suitable
+for any contributor. [good first issue](https://github.com/sigscalr/hyperion/labels/good%20first%20issue)** has extra information to help you make your first contribution. 
+[help wanted](https://github.com/sigscalr/hyperion/labels/help%20wanted)** are issues suitable for someone who isn't a core maintainer and is good to move onto after your first pull request.
+
+Sometimes there won’t be any issues with these labels. That’s ok! There is
+likely still something for you to work on. If you want to contribute but you
+don’t know where to start or can't find a suitable issue, you can 
+[contact us](https://www.sigscalr.io/contact.html)**
+
+Once you see an issue that you'd like to work on, please post a comment saying
+that you want to work on it. Something like "I want to work on this" is fine.
+
+## Ask for Help
+
+
+The best way to reach us with a question when contributing is to ask on:
+
+* The original github issue
+* Our Slack channel : Hyperion public channel**
+
+## Pull Request Lifecycle
+
+1. Open a PR and request reviews only after the full PR is ready for review.
+   - Open *DRAFT* PRs to get feedback on incomplete PRs and let reviewers know what they should be reviewing
+2. Before merging, make sure that all CI/CD tests pass
+   - For larger PRs touching multiple modules, run tests locally using `sigscalr-client`
+   - In addition, for larger PRs, make sure to validate ES compatibility
+   - If any configurations were added or changed, add them to `GETTING_STARTED.md`
+3. Always **squash and merge** any branch merging to `develop`
+4. Always **commit merge** when merging `develop` to `main`
+5. Merged pull request changes will be deployed with the next release.
```

Comment:
3, 4,5 will be different when working on public repos (they can only fork the repo so they won't be able to merge their PR), checkout some guidelines on some big repos like kubernetes, opentelemetry and see how their process is